File: server/src/models/case.ts
import { Column, Entity, PrimaryGeneratedColumn } from "typeorm";
import type { Case } from "../types/case.js";

@Entity("cases")
export default class CaseModel implements Case {
  @PrimaryGeneratedColumn({ type: "bigint" })
  id!: number;

  @Column({ name: "patient_id", type: "bigint" })
  patientId: number;

  @Column({ name: "admission_date", type: "date", nullable: true })
  admissionDate?: Date | null;

  @Column({ name: "discharge_date", type: "date", nullable: true })
  dischargeDate?: Date | null;
}

File: server/models/case.py
from datetime import date
from typing import Optional

from db.database import Base
from sqlalchemy import BigInteger, Date
from sqlalchemy.orm import Mapped, mapped_column


class Case(Base):
    __tablename__ = "cases"

    id: Mapped[int] = mapped_column("id", BigInteger, primary_key=True, autoincrement=True)
    patient_id: Mapped[int] = mapped_column("patient_id", BigInteger, nullable=False)
    admission_date: Mapped[Optional[date]] = mapped_column("admission_date", Date)
    discharge_date: Mapped[Optional[date]] = mapped_column("discharge_date", Date)

    def __repr__(self) -> str:
        return f"<Case id={self.id} patient_id={self.patient_id}>"
